### PR TITLE
Add "cluster restart" and "optimistically confirmed block" to terminology.

### PR DESCRIPTION
**Pull request by wen-coding**
Add two new terminologies:
- cluster restart
- optimistically confirmed block

### CODE REVIEW

File: docs/src/terminology.md [hunk 1]
```diff
@@ -67,6 +67,9 @@ A measure of the network confirmation for the [block](#block).
 
 A set of [validators](#validator) maintaining a single [ledger](#ledger).
 
+## cluster restart
+When there is an outage such that the whole [cluster](#cluster) stalls, human may need to restart most of the [validators](#validator) with a sane state so that the cluster can continue to function. This is different from sporadic single validator restart which does not impact the cluster. See [cluster restart](https://docs.solana.com/running-validator/restart-cluster) for details.
```
Comment (mvines) [inline docs/src/terminology.md:71]: I took a slightly different stab at this:
```suggestion
When a [cluster](#cluster) stops finalizing blocks for a prolonged period of time, as determined by social consensus amongst the validator operators, a pre-defined set of operations is performed by at least 80% of the stake weight to resume block production from the highest optimistically confirmed slot. Any processed but non-confirmed blocks may be discarded during the restart procedure. This is different from sporadic single validator restart which does not impact the cluster. See [cluster restart](https://docs.solana.com/running-validator/restart-cluster) for details.
```

**Review by mvines — COMMENTED** [review]
please wrap at 80 chars 🙏🏼

**Review by mvines — COMMENTED** [review]
please wrap at 80 chars 🙏🏼

File: docs/src/terminology.md [hunk 2]
```diff
@@ -220,6 +228,10 @@ A computer participating in a [cluster](#cluster).
 
 The number of [validators](#validator) participating in a [cluster](#cluster).
 
+## optimistically confirmed block
+A [block](#block) which got the [votes](#vote) from [validators](#validator)
+holding more than 2/3 of total [stake](#stake) in a [cluster](#cluster).
```
Comment (t-nelson) [inline docs/src/terminology.md:233]: i think we need to mention that the votes considered for optimistic confirmation may not have been committed to a block yet.

File: docs/src/terminology.md [hunk 3]
```diff
@@ -67,6 +67,14 @@ A measure of the network confirmation for the [block](#block).
 
 A set of [validators](#validator) maintaining a single [ledger](#ledger).
 
+## cluster restart
+When there is an outage such that the whole [cluster](#cluster) stalls, human
```
Comment (t-nelson) [inline docs/src/terminology.md:71]: i'm not sure this helps much given as we similarly define neither "outage" nor "stall"

File: docs/src/terminology.md (hunk 2 above)
Comment (wen-coding) [reply docs/src/terminology.md:233]: Added. But I suddenly find there is a definition of "confirmed block" above, should I just mark that "we sometimes call it optimistically confirmed"?

Comment (t-nelson) [reply docs/src/terminology.md:233]: maybe just change the `confirmed block` definition to be something like this, given that it is the less precise term?
```
see: [optimistically confirmed block](#optimistically-confirmed-block)
```

**Pull request closed without merging**